UPGRADING.md

```diff
@@ -0,0 +1,183 @@
+# Microsoft Graph PHP SDK Upgrade Guide
+
+This guide highlights backward compatibility breaking changes introduced during major upgrades.
+
+
+## 1.x to 2.0
+
+Version `2.0` highlights:
+- Support for National Clouds
+- Changes in creating a Graph client
+- Changes in configuring your HTTP client (including support for PSR-18 and HTTPlug's HttpAsyncClient implementations)
+- Introducing standardised Graph exception types `GraphClientException` and `GraphServiceException` as more specific `GraphException` types
+- Deprecates support for Guzzle `^6.0`
+- Strongly typed method parameters and return type declarations where possible
+- Psr compliance & other standardisation efforts in request classes and methods
+    - Throwing `Psr\Http\Client\ClientExceptionInterface` instead of `\GuzzleHttp\Exception\GuzzleException` in request methods
+    - Accepting and returning `Psr\Http\Message\StreamInterface` request bodies instead of `GuzzleHttp\Psr7\Stream`
+    - Allow overwriting the default Guzzle client with `Psr\Http\Client\ClientExceptionInterface` for synchronous requests
+    - Allow overwriting the default Guzzle client with HTTPlug's `Http\Client\HttpAsyncClient` for asynchronous requests
+- Introduces a `PageIterator` that pages through a collection response while running a custom callback function against each entity.
+It automatically fetches the nextPage until the end of the collection and allows you to pause and resume processing.
+
+### Support for National Clouds
+We have introduced `NationalCloud` containing Microsoft Graph API endpoint constants to enable you to easily
+set base URLs and in future authenticate against the various supported National Clouds
+
+
+### Creating a Graph client
+- Version 2 deprecates setting HTTP-specific config via methods e.g. `setProxyPort()`
+- Deprecates `setBaseUrl()` and `setApiVersion()` in favour of passing these into the constructor.
+The National Cloud set will be used as the base URL.
```

Comment:
The public cloud endpoint, `https://graph.microsoft.com`,will be set as the default base URL.

Reply:
Updated